### PR TITLE
SW-408 A no result page should not throw an exception

### PR DESCRIPTION
**Pull request by TobiasGraml11**

### CODE REVIEW

File: FinSearchUnified/Bundle/ProductNumberSearch.php [hunk 1]
```diff
@@ -286,10 +286,14 @@ private function fetchSelectedFilterByUserCondition(
 
         $condition = $criteria->getUserCondition($facetName);
 
-        return $this->createSelectedFilter(
-            $criteriaFacet,
-            $condition
-        );
+        if ($condition) {
```
Comment (TheKeymaster) [inline FinSearchUnified/Bundle/ProductNumberSearch.php]: Try to apply [Defensive Coding](https://en.wikipedia.org/wiki/Defensive_programming) in this case. I'd suggest returning null only if there is no condition.

```php
if (!$condition) {
    return null;
}

// Do the usual stuff.
```

Comment (TheKeymaster) [reply FinSearchUnified/Bundle/ProductNumberSearch.php]: Maybe it makes sense to point out in a comment, when a condition may not be able to be fetched.

Comment (TobiasGraml11) [reply FinSearchUnified/Bundle/ProductNumberSearch.php]: Isn't it clear, when the line above is trying to get a `UserCondition`?

Comment (TheKeymaster) [reply FinSearchUnified/Bundle/ProductNumberSearch.php]: That's why I said "maybe". Judge yourself if it is necessary :).

**Pull request merged**